railties/lib/rails/application/finisher.rb

```diff
@@ -124,6 +124,7 @@ def self.complete(_state)
       # the hook are taken into account.
       initializer :set_routes_reloader_hook do |app|
         reloader = routes_reloader
+        reloader.eager_load = app.config.eager_load
```

Comment:
couldn't we use railties eager_load method, instead of passing the config to action pack?
In 3.2, railties could define eager_load! method https://github.com/rails/rails/blob/3-2-stable/railties/lib/rails/engine.rb#L439 and it would get called. Not sure if that was removed or still exists.